### DISCUSSION TITLE
Sphinx - detecting changes in source code

### DISCUSSION THREAD

**Discussion by HalfWhitt** [beeware/toga#2399] [Q&A] [answered]
This project is my first time using Sphinx, so I'm probably missing something here. When I make changes to any of the documentation .rst files, `tox -e docs` detects that and applies changes. When I make changes to the docstings in code, neither docs nor docs-all notices; the only way I've found so far that works is deleting .tox/docs (and/or .tox/docs-all, whichever one I intend to use). There has to be a way to make it detect changes in what it pulls in with autodoc, right?

I also only just discovered the (already installed) sphinx-autobuild (run either directly or from make run), which is *freaking awesome* generally but still only updates on changes to the doc files themselves, not the code.

**Comment by rmartin16** [comment]
I don't have personal knowledge...but found this: https://github.com/sphinx-doc/sphinx-autobuild?tab=readme-ov-file#relevant-sphinx-bugs

In the referenced issue:
> It appears that the underlying issue is at least partially fixed! Now I get automatic rebuilds and browser reloads when I update my code files (as long as the directory they're in is watched). Still no reloads for other static files, but the code files were the biggest pain point since that's where most of the docs live for many of us using autodoc.

I'm not sure, though, how to ensure the relevant directories are watched...

**Reply by rmartin16** [reply, marked as answer]
Oh, ok; I see. I'd never used `sphinx-autobuild` before.
```
sphinx-autobuild -a -j auto ./docs/ ./docs/_build/html/ --watch ./core/src/toga/
```
This is still annoying, though, because it has to rebuild everything...

That said, I'm curious if we could add `tox -e docs-live` to help at least simplify using `sphinx-autobuild`...

**Reply by freakboy3742** [reply]
I definitely wouldn't object to a `docs-live` target...

**Reply by HalfWhitt** [reply]
Thank you! Y'all are both great. : )
